slscore/SLSSrt.cpp

```diff
@@ -219,9 +219,85 @@ int CSLSSrt::libsrt_setup(int port)
     return SLS_OK;
 }
 
+void CSLSSrt::setup_passphrase_endpoint(const char *http_url_passphrase) {
+    if (NULL == http_url_passphrase || strlen(http_url_passphrase) == 0) {
+	    return ;
+	}
+	strcpy(m_http_url_passphrase, http_url_passphrase);
+}
+
+char *CSLSSrt::get_passphrase_endpoint() {
+    return m_http_url_passphrase;
+}
+
+srt_listen_callback_fn handle_srt_listen_callback;
+int handle_srt_listen_callback(void* opaque, SRTSOCKET ns, int hs_version, const struct sockaddr* peeraddr, const char* streamid) {
+    CSLSSrt* tmp = (CSLSSrt*)(opaque);
+    char * pass_url = tmp->get_passphrase_endpoint();
+
+    if (NULL != pass_url && sizeof(pass_url) > 0) {
+        CHttpClient* m_http_client = new CHttpClient;
+
+        char get_passphrase_url[URL_MAX_LEN] = {0};
+        sprintf(get_passphrase_url, "%s?streamid=%s",
+                pass_url, streamid);
+
+        m_http_client->open(get_passphrase_url);
+
```

Comment:
a sleep could be placed here for an initial wait for http response.
i.e 'usleep(500);'